pkg/controller/acmpca/certificateauthoritypermission/controller.go

```diff
@@ -95,17 +95,39 @@ func (e *external) Observe(ctx context.Context, mgd resource.Managed) (managed.E
 	if !ok {
 		return managed.ExternalObservation{}, errors.New(errUnexpectedObject)
 	}
+
+	if meta.GetExternalName(cr) == "" {
+		return managed.ExternalObservation{ResourceExists: false}, nil
+	}
```

Comment:
These checks that assume the resource doesn't exist if we don't have the external name feel a little odd given that we're just deriving our external name from our spec.

Reply:
We could do it without checking for empty string here but then we'd have to use spec fields for identification, which means resource import also has to be done through spec which is not inline with others or handle that in a separate branch.

Reply:
I guess the difference with other resources is that (iirc) we derive the EN from the name automatically before the original observe. 🤔 Would it make sense for us to derive a "temporary" external name from the spec here that we'd use just for observe? e.g. fall back to setting the `principal` and `caARN` to their spec vars?

No need to spend too much time on this though - what we have here is an improvement.